modules/tinymce/src/plugins/table/main/ts/actions/TableActions.ts

```diff
@@ -101,6 +112,63 @@ export const TableActions = (editor: Editor, lazyWire: () => ResizeWire): TableA
 
   const pasteCells = execute(TableOperations.pasteCells, Fun.always, Fun.noop, lazyWire);
 
+  const setTableCellType = (editor: Editor, args: Record<string, any>) =>
+    Obj.get(args, 'type').each((type) => {
+      if (Arr.contains([ 'td', 'th' ], type)) {
+        switchCellType(editor.dom, getCellsFromSelection(editor), type);
+      }
+    });
+
+  const setTableRowType = (editor: Editor, args: Record<string, any>) =>
+    // type: 'header' | 'body' | 'footer'
+    Obj.get(args, 'type').each((type) => {
+      if (Arr.contains([ 'header', 'body', 'footer' ], type)) {
+        Arr.map(getRowsFromSelection(editor), (row) => switchSectionType(editor, row, type));
+      }
+    });
+
+  const makeColumnHeader = execute(TableOperations.makeColumnHeader, Fun.always, Fun.noop, lazyWire);
+  const unmakeColumnHeader = execute(TableOperations.unmakeColumnHeader, Fun.always, Fun.noop, lazyWire);
+
+  const getTableRowType = (editor: Editor) => {
+    const rows = getRowsFromSelection(editor);
+    if (rows.length > 0) {
+      const rowTypes = Arr.map(rows, (r) => getRowType(editor, r));
+      const hasHeader = Arr.contains(rowTypes, 'header');
+      const hasFooter = Arr.contains(rowTypes, 'footer');
+      if (!hasHeader && !hasFooter) {
+        return 'body';
+      } else {
+        const hasBody = Arr.contains(rowTypes, 'body');
+        if (hasHeader && !hasBody && !hasFooter) {
+          return 'header';
+        } else if (!hasHeader && !hasBody && hasFooter) {
+          return 'footer';
+        } else {
+          return '';
+        }
+      }
+    }
+  };
+
+  const getTableCellType = (editor: Editor) => {
+    const cells = getCellsFromSelection(editor);
+    if (cells.length > 0) {
+      const headerCells = Arr.filter(cells, (cell) => Util.getNodeName(cell) === 'th');
```

Comment:
This is almost identical to the `opGetColumnType` logic. Could we move this into Snooker somewhere to remove the duplication?